internal/provider/contract_managment.go

```diff
@@ -0,0 +1,116 @@
+package provider
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/threefoldtech/substrate-client"
+	"github.com/threefoldtech/terraform-provider-grid/pkg/subi"
+)
+
+func IsValidNameContract(sub subi.Substrate, contractID uint64) (bool, error) {
+	contract, err := sub.GetContract(contractID)
+	if errors.Is(err, substrate.ErrNotFound) {
+		return false, nil
+	}
+	if err == nil && !contract.State.IsCreated {
+		return false, nil
+	}
+	if err != nil {
+		return false, errors.Wrapf(err, "couldn't get contract %d info", contractID)
+	}
+	return true, nil
+}
+
+func IsValidCapacityReservationContract(sub subi.Substrate, contractID uint64) error {
+	contract, err := sub.GetContract(contractID)
+	if err != nil {
+		return errors.Wrapf(err, "couldn't get capacity reservation contract %d info", contractID)
+	}
+	if !contract.State.IsCreated {
+		return fmt.Errorf("capacity reservation contract %d not in a created state", contractID)
+	}
+	return nil
+}
+
+func IsValidDeployment(sub subi.Substrate, deploymentID uint64) (bool, error) {
+	_, err := sub.GetDeployment(deploymentID)
+
+	if err != nil && err != substrate.ErrNotFound {
+		return false, errors.Wrapf(err, "couldn't get deployment %d info", deploymentID)
+	}
+	if err == substrate.ErrNotFound {
+		return false, nil
+	}
+	return true, nil
+}
+
+func IsInvalidContract(sub *substrate.Substrate, contractID uint64, deploymentID uint64) (err error) {
```

Comment:
shouldn't `*substrate.Substrate` be changed `subi.Substrate` interface to be compatible with other deployer methods?